app/controllers/mccloud/ChargeAmountReportedController.scala

```diff
@@ -0,0 +1,167 @@
+                                               userAnswersCacheConnector: UserAnswersCacheConnector,
+                                               userAnswersService: UserAnswersService,
+                                               navigator: CompoundNavigator,
+                                               identify: IdentifierAction,
+                                               getData: DataRetrievalAction,
+                                               allowAccess: AllowAccessActionProvider,
+                                               requireData: DataRequiredAction,
+                                               formProvider: ChargeAmountReportedFormProvider,
+                                               val controllerComponents: MessagesControllerComponents,
+                                               config: FrontendAppConfig,
+                                               renderer: Renderer)(implicit ec: ExecutionContext)
+  extends FrontendBaseController
+    with I18nSupport
+    with NunjucksSupport
+    with CommonQuarters
+    with CommonMcCloud {
+
+  private def form(minimumChargeValue: BigDecimal): Form[BigDecimal] = {
+    formProvider(
+      minimumChargeValueAllowed = minimumChargeValue
+    )
+  }
+
+  def onPageLoad(chargeType: ChargeType,
+                 mode: Mode,
+                 srn: String,
+                 startDate: LocalDate,
+                 accessType: AccessType,
+                 version: Int,
+                 index: Index,
+                 schemeIndex: Option[Index]): Action[AnyContent] =
+    (identify andThen getData(srn, startDate) andThen requireData andThen
+      allowAccess(srn, startDate, None, version, accessType)).async { implicit request =>
+      DataRetrievals.retrieveSchemeName { schemeName =>
+        val mininimumChargeValue: BigDecimal = request.sessionData.deriveMinimumChargeValueAllowed
+
+        val preparedForm: Form[BigDecimal] = request.userAnswers
+          .get(ChargeAmountReportedPage(chargeType, index, schemeIndex.map(indexToInt))) match {
+          case Some(value) => form(mininimumChargeValue).fill(value)
+          case None => form(mininimumChargeValue)
+        }
+
+        val viewModel = GenericViewModel(
+          submitUrl = routes.ChargeAmountReportedController.onSubmit(chargeType, mode, srn, startDate, accessType, version, index, schemeIndex).url,
+          returnUrl = controllers.routes.ReturnToSchemeDetailsController.returnToSchemeDetails(srn, startDate, accessType, version).url,
+          schemeName = schemeName
+        )
+
+        val taxQuarterSelection = request.userAnswers.get(TaxQuarterReportedAndPaidPage(chargeType, index, schemeIndex.map(indexToInt)))
+
+        (taxQuarterSelection, lifetimeOrAnnual(chargeType)) match {
+          case (Some(aftQuarter), Some(chargeTypeDesc)) =>
+            val ordinalValue = ordinal(schemeIndex).map(_.resolve).getOrElse("")
```

Comment:
Not sure if we're adopting the same STRING EMPTY val as seen in other repos, not super important

Reply:
I think we don't need this actually because compiler uses a string pool so multiple instances aren't created.